### PR TITLE
feature/KAYAKO-1638

### PR DESCRIPTION
**Pull request by egecankarakasaurea**
JIRA Url: https://jira.devfactory.com/browse/KAYAKO-1638

Coverage before:
coverage_before

Coverage after:
coverage_after

Coverage gain: 115

### CODE REVIEW

**Comment by alexander-a-chernyshev** [comment]
Your Pull Request has been selected for a project-agnostic code review by [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members).

<details><summary>More information</summary>

_This is an initiative to improve overall code quality. If your change is **urgent**, you **don't** have to wait for our review._ 

_NOTE: This review **does not** replace your team's review._

</details>

File: ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/attachmentpreview/AttachmentPreviewFragmentTest.java [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+        LazyHeaders.Builder.class,
+        AttachmentPreviewFragment.class,
+        FileAttachmentUtil.class,
+        AttachmentHelper.class,
+        AttachmentHelper.AttachmentFileType.class
+
+})
+public class AttachmentPreviewFragmentTest {
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
```
Comment (alexander-a-chernyshev) [inline ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/attachmentpreview/AttachmentPreviewFragmentTest.java]: ❗️ This snippet of code contains an instance variable assigned when declared, but not `final`. Consider marking this instance variable as `final` so, it should become a **final** instance variable. This way the code avoids intentionally/accidentally re-assignment of this instance variable and sustain the best OO practices.  

**[More details](https://confluence.devfactory.com/x/OKXNF)**
          
## The solution in this case
```java
@Rule 
public final ExpectedException expectedException = ExpectedException.none();
```
 

<a id="cb451650-4fa9-11e8-b7bf-a93e6485754a" href="none" codereview-uuid-reserved-tag />

File: ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/attachmentpreview/AttachmentPreviewFragmentTest.java [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+        doReturn(mockedView).when(mRoot).findViewById(R.id.ko__button_exit);
+        underTest.onViewCreated(view, bundle);
+        verify(mockedView).setOnClickListener(any(View.OnClickListener.class));
+    }
+
+    @Test
+    public void givenOnActivityCreatedWhenActivityNotKayakoAttachmentPreviewActivityThenThrowIllegalStateException() {
+        expectedException.expect(Exception.class);
+        expectedException.expectMessage("This fragment is meant to be used with KayakoAttachmentPreviewActivity");
+        FragmentActivity activity = mock(FragmentActivity.class);
```
Comment (alexander-a-chernyshev) [inline ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/attachmentpreview/AttachmentPreviewFragmentTest.java]: ❗️ This code is using the class field `activity` as a local variable. Local variables should not shadow class fields. Shadowing fields with a local variable or with a method parameter is a bad practice that reduces code readability, as it is confusing to know whether the field or the variable is being used. Consider renaming this local variable.  

          
## The solution in this case

```JAVA
FragmentActivity activityLocal = mock(FragmentActivity.class);

```
 

<a id="ec8274c0-4fa9-11e8-b7bf-a93e6485754a" href="none" codereview-uuid-reserved-tag />

File: ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/attachmentpreview/AttachmentPreviewFragmentTest.java [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+    }
+
+    @Test
+    public void givenOnActivityCreatedWhenImageUrlNotNullThenSetOnClickListener() {
+        setUpForOnActivityCreated();
+        String imageUrl = "imageUrl";
+        String filePath = null;
+        boolean showSendButton = true;
+        View exitButton = mock(View.class);
+        when(bundle.getString(anyString())).thenReturn(imageUrl, filePath, "sessionId", "userAgent");
```
Comment (alexander-a-chernyshev) [inline ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/attachmentpreview/AttachmentPreviewFragmentTest.java]: ⚠️  This snippet of code uses so-called magic strings and sometimes is difficult to infer their goal in the application. It is a good practice to "extract" magic strings and provide meaningful names for them. In this case, consider extracting them into class-level symbolic constants. This approach increases maintainability (single point of modification) and debugging, avoid duplication and decrease verbosity.  

**[More details](https://confluence.devfactory.com/x/SI8-Ew)**
          
## The solution in this case

```Java  
    private static final String USER_AGENT = "userAgent";
    private static final String SESSION_ID = "sessionId";
    // replace 3 usages of these literals with the new constant
```
 

<a id="1b65a140-4faa-11e8-b7bf-a93e6485754a" href="none" codereview-uuid-reserved-tag />

File: ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/attachmentpreview/AttachmentPreviewFragmentTest.java [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+        when(bundle.containsKey(anyString())).thenReturn(true, true);
+        when(bundle.getString(KayakoAttachmentPreviewActivity.ARG_AGENT_SESSION_ID, null)).thenReturn("sessionId");
+        when(bundle.getString(KayakoAttachmentPreviewActivity.ARG_AGENT_USER_AGENT, null)).thenReturn("userAgent");
+        SessionAuth sessionAuth = Whitebox.<SessionAuth>invokeMethod(underTest, "extractAuthInfo", bundle);
+        Assert.assertNotNull(sessionAuth);
+    }
+
+    @Test
+    public void
+    givenConfigureImageForUrlAttachmentShouldSetImageViewAndLoadingViewVisibleAndAttachmentPlaceHolderGone() throws
```
Comment (alexander-a-chernyshev) [inline ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/attachmentpreview/AttachmentPreviewFragmentTest.java]: ⚠️  This snippet of code contains typos. Spelling errors make reading and understanding the source code harder - mild errors cause the reader to stumble for a fraction of a second, many of them can cause considerable delays; really bad errors can render source code completely unreadable.  

**[More details](https://confluence.devfactory.com/x/QAPvEw)**
          
## The possible solution in this case

```JAVA
@Test 
public void 
givenConfigureImageForUrlAttachmentShouldSetImageViewAndLoadingViewVisibleAndAttachmentPlaceholderGone() throws
                                                                                            ^ lower case

```
 

<a id="61a83b90-4faa-11e8-b7bf-a93e6485754a" href="none" codereview-uuid-reserved-tag />

File: ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/attachmentpreview/AttachmentPreviewFragmentTest.java [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+        View mockedView = mock(View.class);
+        doReturn(mockedView).when(mRoot).findViewById(R.id.ko__button_exit);
+        underTest.onViewCreated(view, bundle);
+        verify(mockedView).setOnClickListener(any(View.OnClickListener.class));
+    }
+
+    @Test
+    public void givenOnActivityCreatedWhenActivityNotKayakoAttachmentPreviewActivityThenThrowIllegalStateException() {
+        expectedException.expect(Exception.class);
+        expectedException.expectMessage("This fragment is meant to be used with KayakoAttachmentPreviewActivity");
```
Comment (alexander-a-chernyshev) [inline ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/attachmentpreview/AttachmentPreviewFragmentTest.java]: ✅ Good to see that exceptional cases are tested as well.

This way the code pass invalid argument/behaviors and check that appropriate exception is raised.

**No change suggested, looks good!**	  

<a id="894f7280-4faa-11e8-b7bf-a93e6485754a" href="none" codereview-uuid-reserved-tag />

**Review by bartosz-rybusinski — APPROVED** [review]

**Pull request merged**